router/router.go

```diff
@@ -1,40 +1,47 @@
 package router
 
 import (
-	"encoding/json"
 	"fmt"
 	"math"
 	"math/rand"
 	"sort"
-	"strconv"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/rudderlabs/rudder-server/router/throttler"
 	"github.com/rudderlabs/rudder-server/services/diagnostics"
-	uuid "github.com/satori/go.uuid"
 
 	"github.com/rudderlabs/rudder-server/config"
 	"github.com/rudderlabs/rudder-server/jobsdb"
-	"github.com/rudderlabs/rudder-server/processor/integrations"
 	"github.com/rudderlabs/rudder-server/rruntime"
 	"github.com/rudderlabs/rudder-server/services/customdestinationmanager"
-	destinationdebugger "github.com/rudderlabs/rudder-server/services/destination-debugger"
 	"github.com/rudderlabs/rudder-server/services/stats"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type WorkerFactory interface {
+	CreateWorkers(rt *HandleT)
+	GetWorker(idx int) WorkerInterface
+	ResetSleep()
+}
+
+type WorkerInterface interface {
+	PostJobOnWorker(job *jobsdb.JobT)
+	WorkerProcess(rt *HandleT)
```

Comment:
Passing rt object here adds complexity I think. We don't why/how it is accessing it. I think it's better to pass actual parameters that's required for that workers. Maybe you can pass some kind of Options/Config object based on the worker in addition to any fixed parameters.